### DISCUSSION TITLE
Image Manipulations not cropping images

### DISCUSSION THREAD

**Discussion by CreativePoweredGeek** [ExpressionEngine/ExpressionEngine#4065] [Feature Requests]
**Description of the problem**
When I have an image manipulation set to CROP to 600px width, it is resizing the entire image as opposed to Cropping.

**How To Reproduce**
In Control Panel, Create a File Group and then set an image manipulation to CROP 600px. Add an image to the File group (in my case it is 1800px x600px) and you will see that the resulting image has been resized as opposed to cropped.

**Error Messages**
None

**Screenshots / Videos / Template Code**

**Environment Details:**
 - Version: EE7.4
 - PHP Version 8.2
 - MySQL Version [latest
 - OS: [Linux]
 - Web Server: [Apache]

**Possible Solution**

**Additional context**

**Comment by intoeetive** [comment]
@CreativePoweredGeek can you share a screenshot of how your image manipulations are set up here?

**Comment by CreativePoweredGeek** [comment]
<img width="1150" alt="Screenshot 2024-02-19 at 10 30 25 AM" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/30536888/cd2da36a-80cb-46de-8b00-5415909ebf9a">

**Comment by CreativePoweredGeek** [comment]
I have also tried with inline Image manipulations and got similar results.

**Comment by intoeetive** [comment]
@CreativePoweredGeek so just to make sure, instead of 2 cropped and 1 resized image, you're getting 3 resized images? Are you using GD or ImageMagick library?

**Comment by CreativePoweredGeek** [comment]
Yes.  That is correct.  And I am using GD

**Comment by CreativePoweredGeek** [comment]
@intoeetive In case it helps,  see attached. 
<img width="1180" alt="Screenshot 2024-02-19 at 10 50 32 AM" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/30536888/452a7bbb-f0e0-4058-960e-88764c9ea2b3">

**Comment by intoeetive** [comment]
Thank you. Also, is this something new in 7.4, or you've had same issue on older versions as well?

**Comment by CreativePoweredGeek** [comment]
To my knowledge it is new.  I'm in the middle of buildign a new site on 7.4 & ran across it.  About ready to update to 7.4.1 ... I'll let you know if anything changes

**Comment by CreativePoweredGeek** [comment]
Updated to 7.4.1 with no change.

**Comment by intoeetive** [comment]
@CreativePoweredGeek it came by surprise to me, probably because I'm using image manipulations often - but that's actually intended behaviour.

The "Crop (part of image)" option actually refers to "resize as close as possible and the crop". This behaviour was like that for a while, if not always.

Unlike on-the-fly `:crop` modifier that is actually cropping only, the manipulation is more similar to `:resize_crop` modifier

**Comment by CreativePoweredGeek** [comment]
OK, So basically,  crop and resize are the same?  That seems counter intuitive… especially when you consider it in the context of image manipulation programs where crop is the method of physically changing size. 

Regarding the :crop on the fly image modifier… it works the sam as the image manipulation you described above?

**Comment by intoeetive** [comment]
No, not the same.

Crop will do the following: First resize down to smallest possible size (greater of height and width), then Crop to desired sizes. So you'll get exact size as requested. Although if you specify just one dimension, the second step will do just nothing, so it will probably function same, or very similar to, resize.

Resize will just resize your image. It will keep the proportions, so one of dimensions might be smaller than requested

`:crop` modifier will JUST crop, without prior resizing.

I agree it's not super intuitive... we are considering making changes in image manipulations as a whole (possibly incorporating some decent third-party solution), but that's not happening very soon. If you have good use cases or examples how we can improve things here, we'd be happy to hear!

I will convert this issue to feature request now

**Reply by CreativePoweredGeek** [reply]
@intoeetive That is not how this is behaving though on a Fresh 7.4 build.  Even when I start with a 2400x600px image, and set a crop manipulation to 600wide by 600 tall, I only get a resize.   Not a crop.

Same goes for :crop on the fly image modifiers.

So just so you know... I went back & looked at a site that began life on ver 5 or so, and has been upgraded ever since and is now on 7.4....  I have simliar "Crop" manipulations (that would have been made prior to 7.4 but can't say for sure when) and they are, in-fact, cropped to square images.  I can provide samples if you need.

**Reply by intoeetive** [reply]
@CreativePoweredGeek do you remember on what major version that sites was at that time? EE 2, 3, 5?

**Reply by CreativePoweredGeek** [reply]
The current live version is on 5.4.3 and I believe it started life as 3.0.   The current 7.4 DEV would have been built of the 5.4.3 version)
